### PR TITLE
Enable GitHub code owner feature for Compilers folder

### PR DESCRIPTION
**Pull request by jcouv**
@jaredpar @jasonmalinowski @dotnet/roslyn-infrastructure  for review.
I could add other folders, let me know.

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# Automatically request reviews when a pull request changes any owned files
+# More information: https://github.com/blog/2392-introducing-code-owners 
+
+src/Compilers/* @dotnet/roslyn-compiler
```
Comment (sharwell) [inline .github/CODEOWNERS]: ❓ Do you need the `*` here? I would have expected the following instead:

```
/src/Compilers/ @dotnet/roslyn-compiler
```

Comment (jcouv) [inline .github/CODEOWNERS]: Possibly, but the doc (linked) gives /docs/* as an example.

Comment (jcouv) [inline .github/CODEOWNERS]: Collection, the example is docs/* (no heading slash).

**Comment by jaredpar** [comment]
Could you add the following mappings as well? 

``` 
build/* @roslyn-infrastructure
src/Setup/* @roslyn-infrastructure 
src/NuGet/* @roslyn-infrastructure
```

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -0,0 +1,8 @@
+# Automatically request reviews when a pull request changes any owned files
+# More information: https://github.com/blog/2392-introducing-code-owners 
+
+build/* @roslyn-infrastructure
```
Comment (sharwell) [inline .github/CODEOWNERS]: ❗️ Group needs to start with `@dotnet/`

Comment (jcouv) [inline .github/CODEOWNERS]: Fixed. Thanks Sam

**Comment by jcouv** [comment]
VS integration test failed on `BasicExpressionEvaluator.EvaluateInvalidExpressions` ([details](https://ci.dot.net/job/dotnet_roslyn/job/master/job/windows_release_vs-integration_prtest/6016/))

FYI @dotnet/roslyn-infrastructure
I'll re-run

**Comment by jcouv** [comment]
test windows_release_vs-integration_prtest please

**Comment by jcouv** [comment]
test windows_release_vs-integration_prtest please

**Comment by jcouv** [comment]
@jasonmalinowski CODEOWNERS change is merged. Go ahead.

**Pull request merged**